src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -143,12 +143,16 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol e
             var semanticFacts = GetSemanticFactsService();
             var semanticModel = context.SemanticModel;
 
-            if (semanticModel.GetTypeInfo(conditionalExpression).Type?.IsValueType == true)
+            var type = semanticModel.GetTypeInfo(conditionalExpression).Type;
+            if (type?.IsValueType == true)
             {
-                // User has something like:  If(str is nothing, nothing, str.Length)
-                // In this case, converting to str?.Length changes the type of this from
-                // int to int?
-                return;
+                if (!(type is INamedTypeSymbol namedType) || namedType.ConstructedFrom.SpecialType != SpecialType.System_Nullable_T)
```

Comment:
I would add a comment explaining that things are ok if the type is actually ```Nullable<SomeType>``` as then there's no conversion problem.